### PR TITLE
[BEAM-815] Fix ApexPipelineOptions conversion and add ApexRunnerRegistrar

### PR DESCRIPTION
**Pull request by iemejia**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by iemejia** [comment]
Just some simple issues I found while trying to execute the some sample pipelines with the Apex runner.
R: @tweise @kennknowles

**Comment by tweise** [comment]
LGTM, will merge it later today unless @kennknowles has further comments.

**Comment by tweise** [comment]
It would be nice if you can reference a JIRA though, in this case BEAM-261 should be sufficient.

**Comment by kennknowles** [comment]
Can you add tests for the registrar? You can see the examples for [FlinkRunner](https://github.com/apache/incubator-beam/blob/master/runners/flink/runner/src/test/java/org/apache/beam/runners/flink/FlinkRunnerRegistrarTest.java), [SparkRunner](https://github.com/apache/incubator-beam/blob/master/runners/spark/src/test/java/org/apache/beam/runners/spark/SparkRunnerRegistrarTest.java), and [DataflowRunner](https://github.com/apache/incubator-beam/blob/master/runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineRegistrarTest.java).

The registrar could also be validated by changing the [pom.xml settings here](https://github.com/apache/incubator-beam/blob/apex-runner/runners/apex/pom.xml#L186) to refer to the runner by short name `TestApexRunner`.

**Comment by iemejia** [comment]
Thanks for the review, I added all your suggestions.
I hope it is ok now.

File: runners/apex/src/test/java/org/apache/beam/runners/apex/ApexRunnerRegistrarTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.apache.beam.runners.apex;
```
Comment (kennknowles) [inline runners/apex/src/test/java/org/apache/beam/runners/apex/ApexRunnerRegistrarTest.java]: RAT needs the Apache header here.

**Comment by kennknowles** [comment]
The Travis failures appear to all be when running `ViewTest`, not related to this PR. Since on `master` Travis gives sketchy signal, I may have been too hasty in ignoring it on the previous Apex PR. It may also be that running embedded Apex with those side input tests hits memory limits of Travis's container infrastructure. @tweise what do you think?

**Comment by tweise** [comment]
Haven't had much luck with integration tests and Travis yet. Only Jenkins gives reliable signals. I'm going to run the tests with the rebased PR locally.

**Comment by iemejia** [comment]
I just fixed the license issue. This time should be ok (modulo the IT issues).

**Comment by tweise** [comment]
integration tests pass locally
LGTM

**Pull request merged**